elements/userlevel/ensuredpdkbuffer.cc

```diff
@@ -88,7 +88,10 @@ EnsureDPDKBuffer::smaction(Packet* p) {
             p->kill();
             return q;
         } else {
-            p->kill();
+            click_chatter("[%s] Could not ensure buffer for packet with length %d bytes. Dropped!", name().c_str(), p->length());
+            if (p->length() <= (DPDKDevice::MBUF_DATA_SIZE - RTE_PKTMBUF_HEADROOM)) {
```

Comment:
I don't understand this? Where the this length computation comes from? What should the user assume?

Reply:
When input frames exceed this length FastClick fails to release the allocated memory (i.e., p->kill() causes seg. fault when it eventually calls the destructor of such a packet). The problem is deeper, this condition just gives a hint on what part of memory allocation might be problematic